File: insertsort.py
#!/usr/bin/env python
# coding=utf-8

def insertsort (A, p, q):
    if (p<q):
        i=p
        while (i<=q):
            key=A[i]
            j=i-1
            while (A[j]>key and j>=p):
                A[j+1]=A[j]
                j-=1
            A[j+1]=key
            i+=1
    else:
        print "sorry"
        
coll = [1,345,5462,43,6,65,2,56,6245,8,3,62,4,2,46,8,4,56,5,95,6,85,624,58,67,7,9,4,6,2,562,7,63,5,764,524,756,624,8,356,73,56,7,43]
for i in coll:
    print i,                                    #python不换行输出
print ""
print "coll after insertsort is ------------------------------------------------------------------------"
insertsort(coll, 0, 43)
for i in coll:
    print i,
print ""

File: insertsort.cpp
#include <iostream>
#include <vector>

using namespace std;
void insertsort(vector<int>& A,const int& p,const int& r)
{
	if(p<r)
	{
		for(int j=p; j<=r;++j)
		{
			int key = A[j];
			int i = j-1;
			while(i>=0 && A[i]>key)
			{
				A[i+1]=A[i];
				--i;
			}
			A[i+1] = key;
		} 
	}
}

int main(int argc,char** argv)
{
	vector<int>coll = {2,3,45,3,54,2,46,658,3,51,65684,6,2457,3,5143,6,5,68,67,243,7789,46,7245,4,3,5,7,67,98};
	for(int i=0; i!=coll.size(); ++i)
		cout<<" "<<coll[i];
	cout<<endl;
	insertsort(coll,0,coll.size()-1);
	cout<<"insertsort------------------------------------"<<endl;
	for(int i=0; i!=coll.size(); ++i)
		cout<<" "<<coll[i];
	cout<<endl;
	return 0;
}
